examples/api/lib/cupertino/date_picker/date_picker.0.dart

```diff
@@ -0,0 +1,163 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Flutter code sample for CupertinoDatePicker
+
+import 'package:flutter/cupertino.dart';
+
+void main() => runApp(const MyApp());
+
+class MyApp extends StatelessWidget {
+  const MyApp({Key? key}) : super(key: key);
+
+  static const String _title = 'Flutter Code Sample';
+
+  @override
+  Widget build(BuildContext context) {
+    return const CupertinoApp(
+      title: _title,
+      home: MyStatelessWidget(),
+    );
+  }
+}
+
+class MyStatelessWidget extends StatefulWidget {
+  const MyStatelessWidget({Key? key}) : super(key: key);
+
+  @override
+  State<MyStatelessWidget> createState() => _MyStatelessWidgetState();
+}
+
+class _MyStatelessWidgetState extends State<MyStatelessWidget> {
+  DateTime date = DateTime(2016, 10, 26);
+  DateTime time = DateTime(2016, 5, 10, 22, 35);
+  DateTime dateTime = DateTime(2016, 8, 3, 17, 45);
+
+  @override
+  Widget build(BuildContext context) {
+    void _buildDialog(Widget child) {
+      showCupertinoModalPopup<void>(
+        context: context,
+        builder: (BuildContext context) => Container(
+          height: 216,
+          padding: const EdgeInsets.only(top: 6.0),
+          margin: EdgeInsets.only(
+            bottom: MediaQuery.of(context).viewInsets.bottom,
+          ),
+          color: CupertinoColors.systemBackground.resolveFrom(context),
+          child: SafeArea(
+            top: false,
+            child: child,
+          ),
+        )
+      );
+    }
+
+    Widget _buildItem({
```

Comment:
Using functions to build part of the widget tree is kind of an "antipattern".  It would be better to split this into a separate `_DatePickerItem` (or some other name) class that returns the resulting widget tree.  It's more performant, since if the widget is marked dirty, only that part rebuilds instead of the entire thing.